packages/core/src/components/graphs/radar.ts

```diff
@@ -0,0 +1,473 @@
+				.attr("transform", `translate(${c.x}, ${c.y})`)
+				.attr("fill", group => colorScale(group.name))
+				.style("fill-opacity", configuration.opacity.selected)
+				.attr("stroke", group => colorScale(group.name))
+				.attr("d", group => oldRadialLineGenerator(group.data))
+				.call(selection => selection
+					.transition(this.services.transitions.getTransition("radar_blobs_enter", animate))
+					.attr("opacity", 1)
+					.attr("d", group => radialLineGenerator(group.data))
+				),
+			update => update
+				.call(selection => selection
+					.transition(this.services.transitions.getTransition("radar_blobs_update", animate))
+					.attr("opacity", 1)
+					.attr("transform", `translate(${c.x}, ${c.y})`)
+					.attr("d", group => radialLineGenerator(group.data))
+				),
+			exit => exit
+				.call(selection => selection
+					.transition(this.services.transitions.getTransition("radar_blobs_exit", animate))
+					.attr("d", group => radialLineGenerator(group.data))
+					.attr("opacity", 0)
+					.remove()
+				)
+		);
+
+		// data dots
+		const dots = DOMUtils.appendOrSelect(this.svg, "g.dots").attr("role", Roles.GROUP);
+		const dotsUpdate = dots.selectAll("circle").data(this.displayDataNormalized);
+		dotsUpdate.join(
+			enter => enter.append("circle").attr("role", Roles.GRAPHICS_SYMBOL),
+			update => update,
+			exit => exit.remove()
+		)
+		.attr("class", d => kebabCase(d.key))
+		.attr("cx", d => polarToCartesianCoords(xScale(d.key), yScale(d.value), c).x)
+		.attr("cy", d => polarToCartesianCoords(xScale(d.key), yScale(d.value), c).y)
+		.attr("r", 0)
+		.attr("opacity", 0)
+		.attr("fill", d => colorScale(d[this.groupMapsTo]));
+
+		// rectangles
+		const xAxesRect = DOMUtils.appendOrSelect(this.svg, "g.x-axes-rect").attr("role", Roles.GROUP);
+		const xAxisRectUpdate = xAxesRect.selectAll("rect").data(this.uniqKeys);
+		xAxisRectUpdate.join(
+			enter => enter.append("rect").attr("role", Roles.GRAPHICS_SYMBOL),
+			update => update,
+			exit => exit.remove()
+		)
+		.attr("x", c.x)
+		.attr("y", c.y - xAxisRectHeight / 2)
+		.attr("width", yScale.range()[1])
+		.attr("height", xAxisRectHeight)
+		.attr("fill", "red")
+		.style("fill-opacity", 0)
+		.attr("transform", key => `rotate(${radToDeg(xScale(key))}, ${c.x}, ${c.y})`);
+
+		// Add event listeners
+		this.addEventListeners();
+
+		oldYScale = yScale; // save the current scale as the old one
+	}
+
+	// append temporarily the label to get the exact space that it occupies
+	labelDimensions = (label: string) => {
+		const tmpTick = DOMUtils.appendOrSelect(this.svg, `g.tmp-tick`);
+		const tmpTickText = DOMUtils.appendOrSelect(tmpTick, `text`).text(label);
+		const { width, height } = DOMUtils.getSVGElementSize(tmpTickText.node(), { useBBox: true });
+		tmpTick.remove();
+		return { width, height };
+	}
+
+	// Given a flat array of objects, if there are missing data on key,
+	// creates corrisponding data with value = 0
+	normalizeFlatData = (dataset: Array<Datum>) => {
```

Comment:
not sure what this method does

Reply:
Sometimes there is missing data, for example:

```
const radarData = [
	{ group: "Sugar", key: "London", value: 25 },
	{ group: "Oil", key: "London", value: 6 },
	{ group: "Water", key: "London", value: 12 },
	{ group: "Sugar", key: "Milan", value: 13 },
	{ group: "Oil", key: "Milan", value: 6 },
	{ group: "Water", key: "Milan", value: 28 },
	{ group: "Sugar", key: "Paris", value: 19 },
	{ group: "Oil", key: "Paris", value: 16 },
	{ group: "Water", key: "Paris", value: 10 },
	{ group: "Sugar", key: "New York", value: 11 },
	{ group: "Oil", key: "New York", value: 18 },
	{ group: "Water", key: "New York", value: 8 },
	{ group: "Sugar", key: "Sydney", value: 12 },
	{ group: "Oil", key: "Sydney", value: 16 }
]
```

In that case there are 3 groups (`Sugar`, `Oil` and `Water`) and 5 keys (`London`, `Milan`, `Paris`, `New York` and `Sydney`) but, as you can see, the datum `{ group: "Water", key: "Sydney", value: xx }` is missing. 
If I compute the path for `Water` blob, I will have a shape composed by 4 vertices and not 5 and it will be a bit strange. So I need to create the missing data and fill them with `value = null`.
`normalizeFlatData` and `normalizeGroupedData` do just that: they fill `data` and `displayData` with missing data, setting `value = null`.

I hope it's clearer now :)

Reply:
It was a design decision to show missing data as `0`, with an `N/A` displayed in the tooltip! Otherwise, "jumping" a datapoint in a radar makes incorrect values being read.